### PR TITLE
Backup: allow to define resources

### PR DESCRIPTION
**Pull request by Smana**
### What problem does this PR solve?

Defining resources for backups is mandatory when you have quotas enabled on a Kubernetes namespace

### What is changed and how does it work?

Adding resources to the "backup" CRD

### CODE REVIEW

**Comment by Yisaer** [comment]
@shuijing198799 @DanielZhangQD @cofyc @weekface  PTAL

File: manifests/backup/backup-s3-br.yaml [hunk 1]
```diff
@@ -5,11 +5,17 @@ metadata:
   name: demo1-backup-s3
   namespace: test1
   # annotations:
-    # iam.amazonaws.com/role: "arn:aws:iam::123456789:role"
+  # iam.amazonaws.com/role: "arn:aws:iam::123456789:role"
```
Comment (DanielZhangQD) [inline manifests/backup/backup-s3-br.yaml:8]: Please revert this change since indent is required here.

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 2]
```diff
@@ -906,6 +906,7 @@ type TiDBAccessConfig struct {
 // +k8s:openapi-gen=true
 // BackupSpec contains the backup specification for a tidb cluster.
 type BackupSpec struct {
+	corev1.ResourceRequirements `json:"resources,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go:909]: Please also help update the `RestoreSpec`, thanks!

File: pkg/backup/backup/backup_manager.go [hunk 3]
```diff
@@ -220,7 +220,8 @@ func (bm *backupManager) makeExportJob(backup *v1alpha1.Backup) (*batchv1.Job, s
 					VolumeMounts: []corev1.VolumeMount{
 						{Name: label.BackupJobLabelVal, MountPath: constants.BackupRootPath},
 					},
-					Env: envVars,
+					Env:       envVars,
```
Comment (DanielZhangQD) [inline pkg/backup/backup/backup_manager.go:223]: Please also update the `backup_cleaner.go` for clean jobs and `restore_manager.go` for restore jobs.

**Comment by Smana** [comment]
@Yisaer @DanielZhangQD I've applied the changes

**Comment by DanielZhangQD** [comment]
@Smana The restore job is not modified, please update that in the restore_manager.go.

**Comment by Smana** [comment]
> @Smana The restore job is not modified, please update that in the restore_manager.go.

Done @DanielZhangQD , sorry I've missed that

**Comment by DanielZhangQD** [comment]
> > @Smana The restore job is not modified, please update that in the restore_manager.go.
> 
> Done @DanielZhangQD , sorry I've missed that

@Smana It's OK. Thanks a lot for the contribution!

**Comment by DanielZhangQD** [comment]
@Smana Please update your branch, thanks!

**Comment by cofyc** [comment]
@Smana can you check this button? our CI bot needs update the branch and test it with the latest code

image

**Comment by Smana** [comment]
@cofyc I've checked the box, sorry for the delay

**Comment by DanielZhangQD** [comment]
@Smana It seems we still cannot edit your branch.
Could you please update your branch?

**Comment by Smana** [comment]
@DanielZhangQD I just rebased my branch

**Comment by cofyc** [comment]
Thanks! @Smana
